src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -394,11 +399,68 @@ public Node getNode() {
 	public void setIndentation(String indentation) {
 		this.indentation = indentation;
 	}
-	
+
 	public String getIndentation() {
 		return indentation;
 	}
 
+	// Experiments API
+
+	@Override
+	public boolean hasExperiment(String featureName) {
+		return Skript.experiments().isUsing(this.getCurrentScript(), featureName);
+	}
+
+
+	@Override
+	public boolean hasExperiment(Experiment experiment) {
+		return Skript.experiments().isUsing(this.getCurrentScript(), experiment);
+	}
+
+	/**
+	 * Marks this as using an experimental feature.
+	 * @param experiment The feature to register.
+	 */
+	@ApiStatus.Internal
+	public void addExperiment(Experiment experiment) {
+		Script script = this.getCurrentScript();
+		@Nullable ExperimentSet set = script.getData(ExperimentSet.class);
+		if (set == null) {
+			set = new ExperimentSet();
+			script.addData(set);
+		}
+		set.add(experiment);
+	}
+
+	/**
+	 * Marks this as no longer using an experimental feature (e.g. during de-registration or reload).
+	 * @param experiment The feature to unregister.
+	 */
+	@ApiStatus.Internal
+	public void removeExperiment(Experiment experiment) {
+		Script script = this.getCurrentScript();
+		@Nullable ExperimentSet set = script.getData(ExperimentSet.class);
+		if (set == null)
+			return;
+		set.remove(experiment);
+	}
+
+	/**
+	 * A snapshot of the experiments this script is currently known to be using.
+	 * This is safe to retain during runtime (e.g. to defer a check) but will
+	 * not see changes, such as if a script subsequently 'uses' another experiment.
+	 *
+	 * @return A snapshot of the current experiment flags in use
+	 */
+	public Experimented experimentSnapshot() {
+		Script script = this.getCurrentScript();
+		@Nullable ExperimentSet set = script.getData(ExperimentSet.class);
+		if (set == null)
+			return new ExperimentSet();
+		else
+			return new ExperimentSet(set);
```

Comment:
```suggestion
		if (set == null)
			return new ExperimentSet();
		return new ExperimentSet(set);
```
```suggestion
		return set != null ? new ExperimentSet(set) : new ExperimentSet();
```
i think either of these formatting options would be better